fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java

```diff
@@ -414,13 +414,11 @@ public long getPrimaryComputeNodeIdByShard(long shardId) throws UserException {
                         String workerAddr = workerInfo.getIpPort();
                         String[] pair = workerAddr.split(":");
 
-                        long backendId = -1L;
-                        if (Config.only_use_compute_node) {
+                        long backendId = GlobalStateMgr.getCurrentSystemInfo()
```

Comment:
need a refactor on `getPrimaryComputeNodeIdByShard` and `getBackendIdsByShard` interface. Looks like the `workaround fix` should be applied to `getPrimaryComputeNodeIdByShard` too, and this repeated retry finding backend id should be applied to `getBackendIdsByShard` as well.